tests/dags/providers/provider_api_scripts/test_metropolitan_museum.py

```diff
@@ -6,181 +6,164 @@
-            "credit_line": "Purchase by subscription, 1879",
-        },
-        "title": "Cover",
-    }
-
-    assert mock_save.call_count == 1
-    for key, value in expected_image.items():
-        assert getattr(actual_image, key) == value
-
-
-def test_get_data_for_image_returns_response_json_with_additional_images(monkeypatch):
-    with open(RESOURCES / "sample_response.json") as f:
-        actual_response_json = json.load(f)
-
-    def mock_get_response_json(query_params, retries=0):
-        return actual_response_json
-
-    monkeypatch.setattr(mma, "_get_response_json", mock_get_response_json)
-    with open(RESOURCES / "sample_additional_image_data.json") as f:
-        image_data = json.load(f)
-
-    r = requests.Response()
-    r.status_code = 200
-    r.json = MagicMock(return_value=image_data)
-    with patch.object(mma.image_store, "add_item", return_value=image_data) as mock_add:
-        mma._get_data_for_image(45734)
-
-    mock_add.assert_called_with(
-        creator="Kiyohara Yukinobu",
-        foreign_identifier="45734-DP251120",
-        foreign_landing_url=(
-            "https://wwwstg.metmuseum.org/art/collection/search/45734"
-        ),
-        image_url="https://images.metmuseum.org/CRDImages/as/original/DP251120.jpg",
-        license_info=(
-            LicenseInfo(
-                "cc0", "1.0", "https://creativecommons.org/publicdomain/zero/1.0/", None
-            )
+@pytest.mark.parametrize(
+    "case_name, response_json, expected",
+    [
+        ("single image", single_object_response, single_expected_data),
+        ("multi image", full_object_response, full_expected_data),
+        (
+            "not cc0",
+            json.loads('{"isPublicDomain": false, "otherData": "is here too"}'),
+            None,
         ),
-        meta_data={
-            "accession_number": "36.100.45",
-            "classification": "Paintings",
-            "culture": "Japan",
-            "date": "late 17th century",
-            "medium": "Hanging scroll; ink and color on silk",
-            "credit_line": (
-                "The Howard Mansfield Collection, Purchase, Rogers Fund, 1936"
-            ),
-        },
-        title="Quail and Millet",
+    ],
+)
+def test_get_record_data_returns_response_json_when_all_ok(
+    case_name, response_json, expected, monkeypatch
+):
+    monkeypatch.setattr(
+        mma.delayed_requester, "get_response_json", lambda x, y: response_json
     )
-
-    assert mock_add.call_count == 3
+    actual = mma.get_record_data(response_json.get("objectID"))
+
+    if expected is None:
+        assert actual is None
+    else:
+        assert len(actual) == len(expected)
+        for i in range(len(actual)):
+            for key, value in expected[i].items():
+                if key == "license_info":
+                    assert actual[i].get(key) == CC0
+                else:
+                    assert actual[i].get(key) == value
```

Comment:
Here's another way to approach this without doing list indexing using [`zip`](https://docs.python.org/3.3/library/functions.html#zip)!

```suggestion
        for actual_result, expected_result in zip(actual, expected):
            for key, value in expected_result.items():
                if key == "license_info":
                    assert actual_result.get(key) == CC0
                else:
                    assert actual_result.get(key) == value
```